### DISCUSSION TITLE
Where can I find the types for Prisma.

### DISCUSSION THREAD

**Discussion by officialabdulrehman** [prisma/prisma#14108] [Q&A]
I am trying to create an abstracted data access layer.

Mongoose dao sample:

```
import { model, Schema, Document, Types, Model } from "mongoose";

import { DAOI } from "./daoI";
import { DTO, DTOCreate, DTOPatch } from "../datamodels/dto";

import { extractFields, flattenObj } from "../util/common";


type TDoc<T extends DTO> = Document & DTOCreate<T>
type ModelType<T extends DTO> = Model<TDoc<T>>


const ID = Types.ObjectId;

export class MongoDAO<TData extends DTO> implements DAOI<TData>{
  protected model: ModelType<TData>
  protected dto: TData = null
  constructor(model: ModelType<TData>, dto: TData) {
    this.model = model;
    this.dto = dto;
  }

  private createDTO(): TData {
    const constructor = this.dto.constructor as any;
    const obj: TData = new constructor();
    return obj;
  }

  public async findById(Id: string, populate: string[] = []): Promise<TData> {
    const id = new ID(Id);
    const result = await this.find({ "_id": id }, 1, 10, populate);
    if (result.data.length == 0) {
      throw new Error(`record with id ${id} not found`);
    }

    const user = result.data[0];
    return user;
  }


}```
how can I achieve this with Prisma and Postgres ?

**Comment by officialabdulrehman** [comment]
Found my answer in issue #5273 and #3929

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
